src/MapperMetadata/MapperMetadata.php

```diff
@@ -241,62 +237,41 @@ private function buildPropertyMapping(): array
         $propertiesMapping = [];
 
         foreach ($this->mappingExtractor->getPropertiesMapping($this) as $propertyMapping) {
+            $transformer = $this->mappingExtractor->resolveTransformer($propertyMapping);
+
+            if (!$transformer) {
+                continue;
+            }
+
+            $propertyMapping->setTransformer($transformer);
```

Comment:
now we've defered the "transformer resolving step", here is how we decide a "property mapping" should be kept.